### ISSUE TITLE
Add GradCache loss

### ISSUE THREAD

**Issue by NohTow** [lightonai/pylate#70]
The [GradCache method](https://arxiv.org/pdf/2101.06983) allows to scale the effective batch size of contrastive loss with a constant requirement in memory, overcoming the issue of gradient accumulation not being equivalent to larger bs for contrastive loss.
I think it would be a nice addition for people training ColBERT models with contrastive and not distillation.

As it is [already implemented in Sentence Transformer](https://github.com/UKPLab/sentence-transformers/blob/master/sentence_transformers/losses/CachedMultipleNegativesRankingLoss.py), adding it to PyLate should be straightforward. 
I think you already experimented with it @raphaelsty, so maybe you can take this one?

cc @tomaarsen

**Comment by raphaelsty** [comment]
I'd be happy to work on this one 🙌

**Comment by NohTow** [comment]
Closing this as done in #98.
